jetty-websocket/websocket-core-common/src/main/java/org/eclipse/jetty/websocket/core/internal/messages/ByteArrayMessageSink.java

```diff
@@ -55,8 +53,8 @@ public void accept(Frame frame, Callback callback)
             long maxBinaryMessageSize = session.getMaxBinaryMessageSize();
             if (maxBinaryMessageSize > 0 && size > maxBinaryMessageSize)
             {
-                throw new MessageTooLargeException(String.format("Binary message too large: (actual) %,d > (configured max binary message size) %,d",
-                    size, maxBinaryMessageSize));
+                callback.failed(new MessageTooLargeException(
```

Comment:
Instead of recalculating the size here (which is also done by the accumulator), why not delegate the max size to the accumulator itself?
At the very least, let the accumulator calculate a running total of the size and ask it here rather than have 2 counts.

Reply:
Also, don't all the previous callbacks need to be failed as well?

Reply:
Yeah I thought I reverted this change, we should throw here so it will be caught and will fail all the callbacks.